### PR TITLE
Update all READMEs to use RC4 MSBuild

### PR DESCRIPTION
**Pull request by kendrahavens**
## Summary

Since the RC4 SDK contains both 1.0 and 1.1 I updated all the links for installing .NET Core to [https://github.com/dotnet/core/blob/master/release-notes/rc4-download.md](https://github.com/dotnet/core/blob/master/release-notes/rc4-download.md) on all the READMEs.

Currently, the only way to download RC4 from dot.net is the 2nd paragraph on the download page: [https://www.microsoft.com/net/download/core](https://www.microsoft.com/net/download/core). Since this paragraph is easy to miss I thought it best to link directly to the GitHub RC4 download table instead of our landing page.

## Reviewers
@MichaelSimons

### CODE REVIEW

File: dotnetapp-prod/README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ dotnetapp-prod Sample
 
 The dotnetapp-prod sample demonstrates how you can build and run the dotnetapp sample using the [.NET Core Runtime Docker image](https://hub.docker.com/r/microsoft/dotnet/). It's a great option for creating a Docker image for production.
 
-The instructions assume that you already have [.NET Core](https://dot.net/core) and [Git](https://git-scm.com/downloads) and [Docker](https://www.docker.com/products/docker) clients installed. They also assume you already know how to target Linux or Windows containers. Do try both image types. You need the latest Windows 10 or Windows Server 2016 to use [Windows containers](http://aka.ms/windowscontainers).
+The instructions assume that you already have [.NET Core](https://dot.net/core)[.NET Core 1.0 RC4](https://github.com/dotnet/core/blob/master/release-notes/rc4-download.md) and [Git](https://git-scm.com/downloads) and [Docker](https://www.docker.com/products/docker) clients installed. They also assume you already know how to target Linux or Windows containers. Do try both image types. You need the latest Windows 10 or Windows Server 2016 to use [Windows containers](http://aka.ms/windowscontainers).
```
Comment (MichaelSimons) [inline dotnetapp-prod/README.md]: Did you intend to leave `[.NET Core](https://dot.net/core)`?  Seems like it should be removed with the addition of the `.NET Core 1.0 RC4` link

File: dotnetapp-dev/README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ dotnetapp-dev Sample
...
 
-The instructions assume that you already have [Git](https://git-scm.com/downloads) and [Docker](https://www.docker.com/products/docker) clients installed. They also assume you already know how to target Linux or Windows containers. Please do try both image types. You need the latest Windows 10 or Windows Server 2016 to use [Windows containers](http://aka.ms/windowscontainers).
+The instructions assume that you already have [Git](https://git-scm.com/downloads), [Docker](https://www.docker.com/products/docker), and [.NET Core 1.1 RC4](https://github.com/dotnet/core/blob/master/release-notes/rc4-download.md) installed. They also assume you already know how to target Linux or Windows containers. Please do try both image types. You need the latest Windows 10 or Windows Server 2016 to use [Windows containers](http://aka.ms/windowscontainers).
```
Comment (MichaelSimons) [inline dotnetapp-dev/README.md]: Do you think this occurrence should be updated to match the formatting of the others?  The others list the 'best' first 😃 - .NET Core

File: dotnetapp-prod/README.md (hunk 1 above)
Comment (kendrahavens) [reply dotnetapp-prod/README.md]: Oh, thank you. I'll take that out!

File: dotnetapp-dev/README.md (hunk 2 above)
Comment (kendrahavens) [reply dotnetapp-dev/README.md]: Sounds good!

**Review by MichaelSimons — APPROVED** [review]

**Comment by kendrahavens** [comment]
Fixes #31 and addresses #21

**Comment by gforceg** [comment]
Is there a page with all prior builds listed for download?
So I can specifically install 1.0.1 on Fedora?

When I try to follow a tutorial, I get errors like this:
```
The specified framework 'Microsoft.NETCore.App', version '1.0.1' was not found.
  - Check application dependencies and target a framework version installed at:
      /opt/dotnet/shared/Microsoft.NETCore.App
  - The following versions are installed:
      1.1.1
      1.1.0
      1.0.0
  - Alternatively, install the framework version '1.0.1'.
```
It would be great if there was a download page with all the builds to date for my OS. Then I could simply download and install it as needed.

**Pull request merged**